fendermint/app/src/app.rs

```diff
@@ -561,22 +595,7 @@ where
 
         let state = match guard.take() {
             Some(state) => state,
-            None => {
-                let db = self.state_store_clone();
-                let state = self.committed_state()?;
-
-                // This would create a partial state, but some client scenarios need the full one.
-                // FvmCheckState::new(db, state.state_root(), state.chain_id())
-                //     .context("error creating check state")?
-
-                FvmExecState::new(
-                    ReadOnlyBlockstore::new(db),
-                    self.multi_engine.as_ref(),
-                    state.block_height.try_into()?,
-                    state.state_params,
-                )
-                .context("error creating check state")?
-            }
+            None => self.create_check_state()?,
```

Comment:
We can avoid loading the 5 blocks state params for `check_tx`, to do this, we must change the `commit` to hold some historical `check_state` instead of purging everything. Not sure if we should do in this PR.